docs/docs/routing.md

```diff
@@ -0,0 +1,261 @@
+---
+id: routing
+title: Routing
+permalink: docs/routing.html
+---
+
+Routing is the process of choosing what to render based on the browser's current URL.
+
+While URLs are an important part of the web, not all apps need to perform routing. For example, single-screen utilities and games will often be unaware of their URL. 
+
+React itself does not have a built-in routing solution. But if you need one, component [state and lifecycle](/react/docs/state-and-lifecycle.html) are all you need to hook React up to the relevant browser APIs. This guide will take you through the process of building a simple Router using these tools.
+
+There are also a number of community projects, such as [react-router](https://github.com/ReactTraining/react-router), that can handle this task for you.
```

Comment:
@ryanflorence 

[This issue](https://github.com/ReactTraining/react-router/issues/4424) has [three](https://github.com/ReactTraining/react-router/issues/4469) [other](https://github.com/ReactTraining/react-router/issues/4497) [issues](https://github.com/ReactTraining/react-router/issues/4519) referencing it. In every single one of them was closed because "that's the way it's going to be".

As for

> can also be implemented in like 10 lines of code on top

that's _exactly_ what you argued for in [the comment](https://github.com/ReactTraining/react-router/issues/4424#issuecomment-276742172) to the props issue. Moreover, that's basically the only argument you give over and over and over and over and over again:

"This is not needed, devs are wrong, you can always implement it yourself with 10 lines of code". Until, obviously, you figure out how to do it and suddenly it's ok, and makes its way into react-router code. 

However. Even with all that this is completely beyond the point.

The main point is: _this is not a react-router guide_. And React team should not endorse a single router solution even if it is named _react_-router. There are multiple other solutions. Some of them better. Some of them worse. Some using completely different approaches to react-router.